app/lib/jsonvalidate.ts

```diff
@@ -0,0 +1,93 @@
+var fs = require('fs');
+var shell = require('shelljs');
+var check = require('validator');
+
+export class JsonValidate {
+	/*
+	 * Checks a json file for correct formatting against some validation function
+	 * @param jsonFilePath path to the json file being validated
+	 * @param jsonValidationFunction function that validates parsed json data against some criteria
+	 * @return the parsed json file
+	 * @throws InvalidDirectoryException if json file doesn't exist, InvalidJsonException on failed parse or *first* invalid field in json
+	*/
+	public validateJson(jsonFilePath: string, jsonValidationFunction: (path: string, jsonData: any) => void, jsonMissingErrorMessage?: string): any {
+		var jsonMissingErrorMsg: string = jsonMissingErrorMessage || 'specified json file does not exist.'
+        this.exists(jsonFilePath, jsonMissingErrorMsg);
+
+        var taskJson;
+        try {
+            taskJson = JSON.parse(fs.readFileSync(jsonFilePath, 'utf8'));
+        }
+        catch (jsonError) {
+            throw new InvalidJsonException("Invalid task json: " + jsonError);
+        }
+        try {
+        	this.validateTask(jsonFilePath, taskJson);
+        }
+        catch (jsonFormatError) {
+			throw new InvalidJsonException("Invalid task json: ", jsonFormatError);
+        }
+        return taskJson;
+	}
+
+	/*
+	 * Wrapper for fs.existsSync that includes a user-specified errorMessage in an InvalidDirectoryException
+	 */
+	public exists(path: string, errorMessage: string) {
+		if(!fs.existsSync(path)) {
+			throw new InvalidDirectoryException(errorMessage);
+		}
+	}
+
+	/*
+     * Validates a parsed json file describing a build task
+     * @param taskPath the path to the original json file
+     * @param taskData the parsed json file
+     */
+	public validateTask(taskPath: string, taskData: any) {
+        var vn = (taskData.name || taskPath);
+
+        if (!taskData.id || !check.isUUID(taskData.id)) {
```

Comment:
still should throw eventually though right?  or should I let users upload an incomplete task and just print these as warnings?